modules/Bio/EnsEMBL/Variation/DBSQL/VariationFeatureAdaptor.pm

```diff
@@ -2414,10 +2415,14 @@ sub fetch_by_spdi_notation{
   my $slice_adaptor = $user_slice_adaptor || $self->db()->dnadb()->get_SliceAdaptor(); 
   my $slice = $slice_adaptor->fetch_by_region('chromosome', $sequence_id ) || $slice_adaptor->fetch_by_region(undef, $sequence_id);  
 
-  # Variation is a substitution  
-  if($check_deleted_seq && $check_inserted_seq_letters && length($deleted_seq) == length($inserted_seq) && length($deleted_seq) == 1){ 
+   
+  my $deleted_seq_length = length($deleted_seq); 
```

Comment:
I might miss something but should we check if the `$deleted_seq` is numeric ?
e.g.:
```my $deleted_seq_length =  ($check_deleted_seq_digit) ? $deleted_seq : length($deleted_seq);```